chia/simulator/full_node_simulator.py

```diff
@@ -1,13 +1,41 @@
-from typing import List, Optional
+import asyncio
+import itertools
+from typing import Collection, List, Optional, Set
 
 from chia.consensus.block_record import BlockRecord
+from chia.consensus.block_rewards import calculate_base_farmer_reward, calculate_pool_reward
 from chia.consensus.multiprocess_validation import PreValidationResult
 from chia.full_node.full_node_api import FullNodeAPI
 from chia.protocols.full_node_protocol import RespondBlock
 from chia.simulator.simulator_protocol import FarmNewBlockProtocol, ReorgProtocol
+from chia.types.blockchain_format.coin import Coin
+from chia.types.blockchain_format.sized_bytes import bytes32
 from chia.types.full_block import FullBlock
 from chia.util.api_decorators import api_request
-from chia.util.ints import uint8
+from chia.util.ints import uint8, uint32, uint64
+from chia.wallet.transaction_record import TransactionRecord
+from chia.wallet.util.wallet_types import AmountWithPuzzlehash
+from chia.wallet.wallet import Wallet
+
+
+async def wait_for_coins_in_wallet(coins: Set[Coin], wallet: Wallet):
```

Comment:
I'm a bit conflicted because this function will be useful in the same places where the full node wrapper is, yes?  I don't mean this is the right place, but having one thing in the chia package and another in the tests package doesn't obviously make sense to me yet.